implementations/scripts/SourceCode.py

```diff
@@ -0,0 +1,101 @@
+class IsSourceCode:
+
+    def __init__(self, source_code_exclude_list=None, method="naive"):
+        """
+        A IsSourceCode object is used to help define source code.
+        :param source_code_exclude_list: a list of either
+            file extensions or directories to exclude
+        :param method: a string which can be "naive",
+            "folder_exclude" or "extension_exclude"
+        For example,
+        source_code_exclude_list = ["py", "other", "gitignore"]
+        or source_code_exclude_list = ["tests/", "bin/", "drivers/base/"]
+        """
+        self.source_code_exclude_list = source_code_exclude_list
+        self.method = method
+
+    def is_source_code(self, commit):
```

Comment:
Since the class is already `IsSourceCode`, this can be, for example, `check()`.

Reply:
Oops! will change that.